src/fs_clonefile.cpp

```diff
@@ -67,24 +67,24 @@ writen(const int     fd,
           return -1;
         }
 
-      nleft -= nwritten;
-      buf   += nwritten;
+      nleft -= (size_t)nwritten;
+      buf   += (size_t)nwritten;
     }
 
-  return count;
+  return nwritten;
```

Comment:
I think of it as a data pipe.  The OS gives us a total of the bytes written and then we hand that back to FUSE to deliver to the application.  If anything goes wrong along the way then returning a fixed value might lead to odd things, especially for a partial write (it's not a fail because data changed, but it's not a success because the whole request wasn't completed).  That'd be my only concern.